tools/crm_error.c

```diff
@@ -132,13 +133,21 @@ main(int argc, char **argv)
 
         free(longest_rc_str);
 
-        // 256 is a hacky magic number that "should" be enough
-        if (options.result_type == pcmk_result_pcmk_rc) {
-            start = pcmk_rc_error - 256;
-            end = PCMK_CUSTOM_OFFSET;
-        } else {
-            start = 0;
-            end = 256;
+        switch (options.result_type) {
+            case pcmk_result_legacy:
+                start = pcmk_ok;
+                end = pcmk_err_unknown_format + 1;
+                break;
+            case pcmk_result_pcmk_rc:
+                start = pcmk_rc_error - pcmk__n_rc + 1;
+                end = 256;  // should be enough for almost any system error code
+                break;
+            case pcmk_result_crm_exitcode:
+                start = CRM_EX_OK;
+                end = CRM_EX_MAX + 1;
+                break;
+            default:    // Make the compiler happy
+                break;
```

Comment:
I wonder if this is worth abstracting too e.g. pcmk_result_bounds(result_kind, &start, &end)